models/DataObject/Service.php

```diff
@@ -1698,16 +1697,18 @@ public static function getCsvDataForObject(Concrete $object, string $requestedLa
     public static function getCsvData(string $requestedLanguage, LocaleServiceInterface $localeService, Listing $list, array $fields, string $header = '', bool $addTitles = true, array $context = []): array
     {
         $data = [];
-        Logger::debug('objects in list:' . count($list->getObjects()));
+        Logger::debug('objects in list:' . $list->getCount());
 
         if (class_exists(GridData\DataObject::class)) {
             $helperDefinitions = GridData\DataObject::getHelperDefinitions();
         } else {
             $helperDefinitions = self::getHelperDefinitions();
         }
 
-        foreach ($list->getObjects() as $object) {
-            if ($fields) {
+        $objects = $list->getObjects();
+
+        if ($fields) {
```

Comment:
Can be moved to the if block, right? Maybe also the $helperDefinitions loadings.
```suggestion
        if ($fields) {
            $objects = $list->getObjects();
```